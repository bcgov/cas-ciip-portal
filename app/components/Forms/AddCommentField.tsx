import React, {useState} from 'react';
import {FieldProps} from '@rjsf/core';
import {Form, Button} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faCommentAlt} from '@fortawesome/free-solid-svg-icons';
import {toast} from 'react-toastify';

const AddCommentField: React.FunctionComponent<FieldProps<string>> = (
  props
) => {
  const [expanded, setExpanded] = useState(false);
  const [value, setValue] = useState(props.formData);
  const title = props.formData ? 'Edit comment' : 'Add a comment';

  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setValue(e.target.value);
  };

  const handleCancel = () => {
    setValue(props.formData);
    setExpanded(false);
  };

  const handleSave = () => {
    props.onChange(value);
    setExpanded(false);
    toast.success(
      'Your comment was saved for review upon application submission',
      {
        autoClose: 5000,
        position: 'top-center'
      }
    );
  };

  return (
    <>
      <style jsx>{`
        .title,
        .report-field .controls {
          display: flex;
          width: 100%;
          justify-content: flex-end;
          align-items: center;
        }

        .report-field .controls {
          margin-top: 16px;
        }

        .report-field {
          background-color: #fcf8e2;
          border: 1px solid #ccc;
          border-radius: 4px;
          padding: 16px;
          margin-top: 16px;
        }
      `}</style>
      <span className="title">
        <Button variant="link" onClick={() => setExpanded(true)}>
          <FontAwesomeIcon icon={faCommentAlt} />
          &nbsp;
          {title}
        </Button>
      </span>
      {expanded && (
        <div className="report-field">
          <legend>CIIP Application Comment</legend>

          <p>
            This area is only for leaving a comment regarding the data that you
            reported on this page. This comment will be visible by the person
            reviewing this CIIP application once the application is submitted.
          </p>
          <p>
            For any technical issue, or to review documentation, please click on
            the help bubble in the bottom right of your screen.
          </p>
          <Form.Control
            as="textarea"
            rows={8}
            value={value}
            onChange={handleChange}
          />
          <div className="controls">
            <Button variant="outline-dark" onClick={handleCancel}>
              Cancel
            </Button>
            <Button
              style={{marginLeft: 16}}
              variant="success"
              onClick={handleSave}
            >
              Save Comment
            </Button>
          </div>
        </div>
      )}
    </>
  );
};

export default AddCommentField;
