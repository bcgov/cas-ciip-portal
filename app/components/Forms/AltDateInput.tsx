// https://github.com/rjsf-team/react-jsonschema-form/blob/v1.8.1/src/components/widgets/AltDateWidget.js

import React, { Component } from "react";
import PropTypes from "prop-types";

function rangeOptions(start, stop) {
  let options = [];
  for (let i = start; i <= stop; i++) {
    options.push({ value: i, label: pad(i, 2) });
  }
  return options;
}

function readyForChange(state) {
  return Object.keys(state).every(key => state[key] !== -1);
}

function DateElement(props) {
  const {
    type,
    range,
    value,
    select,
    rootId,
    disabled,
    readonly,
    autofocus,
    registry,
    onBlur,
  } = props;
  const id = rootId + "_" + type;
  const { SelectWidget } = registry.widgets;
  return (
    <SelectWidget
      schema={{ type: "integer" }}
      id={id}
      className="form-control"
      options={{ enumOptions: rangeOptions(range[0], range[1]) }}
      placeholder={type}
      value={value}
      disabled={disabled}
      readonly={readonly}
      autofocus={autofocus}
      onChange={value => select(type, value)}
      onBlur={onBlur}
    />
  );
}

class AltDateInput extends Component {
  static defaultProps = {
    time: false,
    disabled: false,
    readonly: false,
    autofocus: false,
    options: {
      yearsRange: [1900, new Date().getFullYear() + 2],
    },
  };

  constructor(props) {
    super(props);
    this.state = parseDateString(props.value, props.time);
  }

  componentWillReceiveProps(nextProps) {
    this.setState(parseDateString(nextProps.value, nextProps.time));
  }

  shouldComponentUpdate(nextProps, nextState) {
    return shouldRender(this, nextProps, nextState);
  }

  onChange = (property, value) => {
    this.setState(
      { [property]: typeof value === "undefined" ? -1 : value },
      () => {
        // Only propagate to parent state if we have a complete date{time}
        if (readyForChange(this.state)) {
          this.props.onChange(toDateString(this.state, this.props.time));
        }
      }
    );
  };

  setNow = event => {
    event.preventDefault();
    const { time, disabled, readonly, onChange } = this.props;
    if (disabled || readonly) {
      return;
    }
    const nowDateObj = parseDateString(new Date().toJSON(), time);
    this.setState(nowDateObj, () => onChange(toDateString(this.state, time)));
  };

  clear = event => {
    event.preventDefault();
    const { time, disabled, readonly, onChange } = this.props;
    if (disabled || readonly) {
      return;
    }
    this.setState(parseDateString("", time), () => onChange(undefined));
  };

  get dateElementProps() {
    const { time, options } = this.props;
    const { year, month, day, hour, minute, second } = this.state;
    const data = [
      {
        type: "year",
        range: options.yearsRange,
        value: year,
      },
      { type: "month", range: [1, 12], value: month },
      { type: "day", range: [1, 31], value: day },
    ];
    if (time) {
      data.push(
        { type: "hour", range: [0, 23], value: hour },
        { type: "minute", range: [0, 59], value: minute },
        { type: "second", range: [0, 59], value: second }
      );
    }
    return data;
  }

  render() {
    const {
      id,
      disabled,
      readonly,
      autofocus,
      registry,
      onBlur,
      options,
    } = this.props;
    return (
      <ul className="list-inline">
        {this.dateElementProps.map((elemProps, i) => (
          <li key={i}>
            <DateElement
              rootId={id}
              select={this.onChange}
              {...elemProps}
              disabled={disabled}
              readonly={readonly}
              registry={registry}
              onBlur={onBlur}
              autofocus={autofocus && i === 0}
            />
          </li>
        ))}
        {(options.hideNowButton !== "undefined"
          ? !options.hideNowButton
          : true) && (
          <li>
            <a href="#" className="btn btn-info btn-now" onClick={this.setNow}>
              Now
            </a>
          </li>
        )}
        {(options.hideClearButton !== "undefined"
          ? !options.hideClearButton
          : true) && (
          <li>
            <a
              href="#"
              className="btn btn-warning btn-clear"
              onClick={this.clear}>
              Clear
            </a>
          </li>
        )}
      </ul>
    );
  }
}

if (process.env.NODE_ENV !== "production") {
  AltDateInput.propTypes = {
    schema: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    value: PropTypes.string,
    required: PropTypes.bool,
    disabled: PropTypes.bool,
    readonly: PropTypes.bool,
    autofocus: PropTypes.bool,
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    time: PropTypes.bool,
    options: PropTypes.object,
  };
}

export default AltDateInput;



function shouldRender(comp, nextProps, nextState) {
  const { props, state } = comp;
  return !deepEquals(props, nextProps) || !deepEquals(state, nextState);
}

function isArguments(object) {
  return Object.prototype.toString.call(object) === "[object Arguments]";
}

function parseDateString(dateString, includeTime = true) {
  if (!dateString) {
    return {
      year: -1,
      month: -1,
      day: -1,
      hour: includeTime ? -1 : 0,
      minute: includeTime ? -1 : 0,
      second: includeTime ? -1 : 0,
    };
  }
  const date = new Date(dateString);
  if (Number.isNaN(date.getTime())) {
    throw new Error("Unable to parse date " + dateString);
  }
  return {
    year: date.getUTCFullYear(),
    month: date.getUTCMonth() + 1, // oh you, javascript.
    day: date.getUTCDate(),
    hour: includeTime ? date.getUTCHours() : 0,
    minute: includeTime ? date.getUTCMinutes() : 0,
    second: includeTime ? date.getUTCSeconds() : 0,
  };
}

function toDateString(
  { year, month, day, hour = 0, minute = 0, second = 0 },
  time = true
) {
  const utcTime = Date.UTC(year, month - 1, day, hour, minute, second);
  const datetime = new Date(utcTime).toJSON();
  return time ? datetime : datetime.slice(0, 10);
}

function pad(num, size) {
  let s = String(num);
  while (s.length < size) {
    s = "0" + s;
  }
  return s;
}

function deepEquals(a, b, ca = [], cb = []) {
  // Partially extracted from node-deeper and adapted to exclude comparison
  // checks for functions.
  // https://github.com/othiym23/node-deeper
  if (a === b) {
    return true;
  } else if (typeof a === "function" || typeof b === "function") {
    // Assume all functions are equivalent
    // see https://github.com/mozilla-services/react-jsonschema-form/issues/255
    return true;
  } else if (typeof a !== "object" || typeof b !== "object") {
    return false;
  } else if (a === null || b === null) {
    return false;
  } else if (a instanceof Date && b instanceof Date) {
    return a.getTime() === b.getTime();
  } else if (a instanceof RegExp && b instanceof RegExp) {
    return (
      a.source === b.source &&
      a.global === b.global &&
      a.multiline === b.multiline &&
      a.lastIndex === b.lastIndex &&
      a.ignoreCase === b.ignoreCase
    );
  } else if (isArguments(a) || isArguments(b)) {
    if (!(isArguments(a) && isArguments(b))) {
      return false;
    }
    let slice = Array.prototype.slice;
    return deepEquals(slice.call(a), slice.call(b), ca, cb);
  } else {
    if (a.constructor !== b.constructor) {
      return false;
    }

    let ka = Object.keys(a);
    let kb = Object.keys(b);
    // don't bother with stack acrobatics if there's nothing there
    if (ka.length === 0 && kb.length === 0) {
      return true;
    }
    if (ka.length !== kb.length) {
      return false;
    }

    let cal = ca.length;
    while (cal--) {
      if (ca[cal] === a) {
        return cb[cal] === b;
      }
    }
    ca.push(a);
    cb.push(b);

    ka.sort();
    kb.sort();
    for (var j = ka.length - 1; j >= 0; j--) {
      if (ka[j] !== kb[j]) {
        return false;
      }
    }

    let key;
    for (let k = ka.length - 1; k >= 0; k--) {
      key = ka[k];
      if (!deepEquals(a[key], b[key], ca, cb)) {
        return false;
      }
    }

    ca.pop();
    cb.pop();

    return true;
  }
}