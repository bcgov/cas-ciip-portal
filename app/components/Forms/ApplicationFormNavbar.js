import React from 'react';
import {Nav} from 'react-bootstrap';
import {createFragmentContainer} from 'react-relay';
import Link from 'next/link';

const ApplicationFormNavbarComponent = props => {
  const {wizard, application} = props;
  return (
    <>
      <Nav justify variant="tabs" defaultActiveKey="/home">
        {wizard.edges.map(({node}) => (
          <Nav.Item key={node.formJsonByFormId.id}>
            <Link
              passHref
              href={{
                pathname: '/ciip-application',
                query: {
                  formId: node.formJsonByFormId.id,
                  applicationId: application.id
                }
              }}
            >
              <Nav.Link>{node.formJsonByFormId.name}</Nav.Link>
            </Link>
          </Nav.Item>
        ))}
      </Nav>
    </>
  );
};

export default createFragmentContainer(ApplicationFormNavbarComponent, {});
