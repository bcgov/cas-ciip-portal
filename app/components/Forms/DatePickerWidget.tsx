import React, {useState} from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import {WidgetProps} from 'react-jsonschema-form';
import {dateTimeFormat} from 'functions/formatDates';

function getDateString(date) {
  return dateTimeFormat(date, 'date_year_first');
}

const DatePickerWidget: React.FunctionComponent<WidgetProps> = ({
  id,
  value,
  disabled,
  readonly,
  onBlur,
  onChange,
  onFocus
}) => {
  const [day, setDay] = useState(value ? new Date(value) : undefined);

  const handleChange = (d) => {
    setDay(d);
    onChange(getDateString(d));
  };

  const handleBlur = () => {
    onBlur(id, getDateString(day));
  };

  const handleFocus = () => {
    onFocus(id, getDateString(day));
  };

  return (
    <div>
      <DatePicker
        disabled={disabled}
        readOnly={readonly}
        className="form-control"
        selected={day}
        onChange={handleChange}
        onBlur={handleBlur}
        onFocus={handleFocus}
        dateFormat="yyyy-MM-dd"
        placeholderText="YYYY-MM-DD"
        showMonthDropdown
        showYearDropdown
      />
    </div>
  );
};

export default DatePickerWidget;
