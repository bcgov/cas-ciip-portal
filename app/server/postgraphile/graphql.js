const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
