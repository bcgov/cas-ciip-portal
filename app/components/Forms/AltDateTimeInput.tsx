// https://github.com/rjsf-team/react-jsonschema-form/blob/v1.8.1/src/components/widgets/AltDateTimeWidget.js

import React from "react";
import PropTypes from "prop-types";
import AltDateInput from "./AltDateInput";
import { WidgetProps } from "react-jsonschema-form";

function AltDateTimeInput(props: WidgetProps) {
  const { AltDateInput } = props.registry.widgets;
  return <AltDateInput time {...props} />;
}

if (process.env.NODE_ENV !== "production") {
  AltDateTimeInput.propTypes = {
    schema: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    value: PropTypes.string,
    required: PropTypes.bool,
    onChange: PropTypes.func,
    options: PropTypes.object,
  };
}

AltDateTimeInput.defaultProps = {
  ...AltDateInput.defaultProps,
  time: true,
};

export default AltDateTimeInput;