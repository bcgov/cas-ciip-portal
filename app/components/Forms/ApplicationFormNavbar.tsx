import React from 'react';
import {Nav} from 'react-bootstrap';
import {createFragmentContainer, graphql} from 'react-relay';
import Link from 'next/link';
import {ApplicationFormNavbar_applicationRevision} from 'ApplicationFormNavbar_applicationRevision.graphql';
import ApplicationPage from 'pages/reporter/application/[applicationId]';

interface Props {
  applicationRevision: ApplicationFormNavbar_applicationRevision;
  selectedFormId: string;
  confirmationPage: boolean;
}

const ApplicationFormNavbarComponent: React.FunctionComponent<Props> = ({
  applicationRevision,
  selectedFormId,
  confirmationPage
}) => {
  return (
    <div className="nav-guide-container">
      <Nav justify className="nav-guide" variant="pills">
        {applicationRevision.orderedFormResults.edges.map(({node}) => (
          <Nav.Item key={node.id}>
            <Link
              passHref
              href={ApplicationPage.getRoute(
                applicationRevision.applicationByApplicationId.id,
                node.formJsonByFormId.id
              )}
            >
              <Nav.Link active={node.id === selectedFormId}>
                {node.formJsonByFormId.name}{' '}
              </Nav.Link>
            </Link>
          </Nav.Item>
        ))}
        <Nav.Item key="submit">
          <Link
            passHref
            href={ApplicationPage.getRoute(
              applicationRevision.applicationByApplicationId.id,
              undefined,
              true
            )}
          >
            <Nav.Link active={confirmationPage}>Summary</Nav.Link>
          </Link>
        </Nav.Item>
        <style jsx global>{`
          .nav-guide-container {
            border: 1px solid #bbb;
            padding: 30px;
            background: #eee;
            border-radius: 6px;
            margin-bottom: 36px;
          }
          .nav-guide.nav-pills {
            border: 1px solid #00336675;
            border-radius: 5px;
          }

          .nav-guide.nav-pills .nav-item {
            display: flex;
            justify-content: center;
            align-items: center;
            background: white;
            border-left: 1px solid #0369;
          }

          .nav-guide.nav-pills .nav-item a {
            color: #036;
          }

          .nav-guide.nav-pills .nav-item a.active {
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            border-radius: 0;
            background: #036;
            color: white;
          }
        `}</style>
      </Nav>
    </div>
  );
};

export default createFragmentContainer(ApplicationFormNavbarComponent, {
  applicationRevision: graphql`
    fragment ApplicationFormNavbar_applicationRevision on ApplicationRevision {
      applicationByApplicationId {
        id
      }
      orderedFormResults {
        edges {
          node {
            id
            formJsonByFormId {
              id
              name
            }
          }
        }
      }
    }
  `
});
