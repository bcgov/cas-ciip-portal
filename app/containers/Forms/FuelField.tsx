import React from 'react';
import {FieldProps} from '@rjsf/core';
import {createFragmentContainer, graphql} from 'react-relay';
import {FuelField_query} from 'FuelField_query.graphql';
import ObjectField from '@rjsf/core/dist/cjs/components/fields/ObjectField';

interface Props extends FieldProps {
  query: FuelField_query;
}

const FuelField: React.FunctionComponent<Props> = (props) => {
  const {formData, query, onChange} = props;

  const handlefuelChange = (fuelRowId: number) => {
    const fuel = query.allFuels.edges.find(({node}) => node.rowId === fuelRowId)
      ?.node;
    onChange({
      ...formData,
      fuelRowId,
      fuelUnits: fuel?.units
    });
  };

  const handleChange = (fuel) => {
    if (formData.fuelRowId === fuel.fuelRowId) onChange(fuel);
    else handlefuelChange(fuel.fuelRowId);
  };

  return <ObjectField {...props} onChange={handleChange} />;
};

export default createFragmentContainer(FuelField, {
  query: graphql`
    fragment FuelField_query on Query {
      allFuels(condition: {state: "active"}) {
        edges {
          node {
            units
            rowId
          }
        }
      }
    }
  `
});
