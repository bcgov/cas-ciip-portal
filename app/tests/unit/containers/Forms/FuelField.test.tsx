import { FuelFieldComponent } from "containers/Forms/FuelField";
import { shallow } from "enzyme";
import React from "react";
import { createDefaultJsonSchemaFormProps } from "tests/json-schema-utils";
import { FuelField_query } from "__generated__/FuelField_query.graphql";

describe("The Fuel Field", () => {
  it("renders an interactive form when the form data fuel is active", () => {
    const testQuery: FuelField_query = {
      allFuels: {
        edges: [
          {
            node: {
              rowId: 1,
              state: "published",
              units: "unit",
            },
          },
        ],
      },
      " $refType": "FuelField_query",
    };
    const initialProps = {
      ...createDefaultJsonSchemaFormProps(),
      formData: {
        fuelRowId: 1,
      },
      query: testQuery,
    };

    const component = shallow(<FuelFieldComponent {...initialProps} />);

    expect(component).toMatchSnapshot();
    expect(component.find("ObjectField").at(0).prop("disabled")).toBe(false);
  });

  it("renders an interactive form when the form data is undefined", () => {
    const testQuery: FuelField_query = {
      allFuels: {
        edges: [
          {
            node: {
              rowId: 1,
              state: "published",
              units: "unit",
            },
          },
        ],
      },
      " $refType": "FuelField_query",
    };
    const initialProps = {
      ...createDefaultJsonSchemaFormProps(),
      formData: {
        fuelRowId: undefined,
      },
      query: testQuery,
    };

    const component = shallow(<FuelFieldComponent {...initialProps} />);

    expect(component).toMatchSnapshot();
    expect(component.find("ObjectField").at(0).prop("disabled")).toBe(false);
  });

  it("renders a disabled form when the form data fuel is archived", () => {
    const testQuery: FuelField_query = {
      allFuels: {
        edges: [
          {
            node: {
              rowId: 1,
              state: "archived",
              units: "unit",
            },
          },
        ],
      },
      " $refType": "FuelField_query",
    };
    const initialProps = {
      ...createDefaultJsonSchemaFormProps(),
      formData: {
        fuelRowId: 1,
      },
      query: testQuery,
    };

    const component = shallow(<FuelFieldComponent {...initialProps} />);

    expect(component).toMatchSnapshot();
    expect(component.find("ObjectField").at(0).prop("disabled")).toBe(true);
  });
});
