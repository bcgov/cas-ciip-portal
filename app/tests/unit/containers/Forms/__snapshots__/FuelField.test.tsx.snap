// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Fuel Field renders an alert box and a disabled form when the form data fuel is archived 1`] = `
<Fragment>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     This fuel has been archived. Please remove it and select an appropriate alternative.
  </Alert>
  <ObjectField
    autofocus={false}
    disabled={true}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "fuelRowId": 1,
      }
    }
    idSchema={
      Object {
        "$id": null,
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "FuelField_query",
        "allFuels": Object {
          "edges": Array [
            Object {
              "node": Object {
                "rowId": 1,
                "state": "archived",
                "units": "unit",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The Fuel Field renders an interactive form when the form data fuel is active 1`] = `
<Fragment>
  <ObjectField
    autofocus={false}
    disabled={false}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "fuelRowId": 1,
      }
    }
    idSchema={
      Object {
        "$id": null,
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "FuelField_query",
        "allFuels": Object {
          "edges": Array [
            Object {
              "node": Object {
                "rowId": 1,
                "state": "published",
                "units": "unit",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The Fuel Field renders an interactive form when the form data is undefined 1`] = `
<Fragment>
  <ObjectField
    autofocus={false}
    disabled={false}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "fuelRowId": undefined,
      }
    }
    idSchema={
      Object {
        "$id": null,
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "FuelField_query",
        "allFuels": Object {
          "edges": Array [
            Object {
              "node": Object {
                "rowId": 1,
                "state": "published",
                "units": "unit",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;
